frontend/packages/dev-console/src/components/topology/componentUtils.ts

```diff
@@ -170,10 +176,35 @@ const groupWorkoadDropTargetSpec: DropTargetSpec<
   }),
 };
 
+const graphEventSourceDropTargetSpec: DropTargetSpec<
+  GraphElement,
+  any,
+  { droppable: boolean; canDrop: boolean; dropTarget: boolean; edgeDragging: boolean },
+  NodeProps
+> = {
+  accept: [MOVE_EV_SRC_CONNECTOR_DROP_TYPE],
+  canDrop: (item, monitor, props) => {
+    return isEdge(item) ? item.getSource() !== props.element : false;
```

Comment:
Once `item` is typed as `Edge` we can omit the `isEdge` check:
```suggestion
    return item.getSource() !== props.element;
```